client/modules/IDE/pages/IDEView.jsx

```diff
@@ -78,347 +79,256 @@ function WarnIfUnsavedChanges() {
 
 export const CmControllerContext = React.createContext({});
 
-class IDEView extends React.Component {
-  constructor(props) {
-    super(props);
-    this.handleGlobalKeydown = this.handleGlobalKeydown.bind(this);
-
-    this.state = {
-      consoleSize: props.ide.consoleIsExpanded ? 150 : 29,
-      sidebarSize: props.ide.sidebarIsExpanded ? 160 : 20
-    };
-  }
+const IDEView = (props) => {
+  const ide = useSelector((state) => state.ide);
+  const selectedFile = useSelector(selectActiveFile);
+  const preferences = useSelector((state) => state.preferences);
+  const project = useSelector((state) => state.project);
+  const isUserOwner = useSelector(getIsUserOwner);
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
 
-  componentDidMount() {
-    // If page doesn't reload after Sign In then we need
-    // to force cleared state to be cleared
-    this.props.clearPersistedState();
+  const [consoleSize, setConsoleSize] = useState(
+    ide.consoleIsExpanded ? 150 : 29
+  );
+  const [sidebarSize, setSidebarSize] = useState(
+    ide.sidebarIsExpanded ? 160 : 20
+  );
+  const rootFile = useSelector(selectRootFile);
+  const canEditProject = useSelector(selectCanEditSketch);
 
-    this.props.stopSketch();
-    if (this.props.params.project_id) {
-      const { project_id: id, username } = this.props.params;
-      if (id !== this.props.project.id) {
-        this.props.getProject(id, username);
-      }
-    }
+  const cmRef = useRef({});
+  let overlay = null;
```

Comment:
The overlay will need to be `useRef` as well.

Although…it seems like the current code is only using this variable to manipulate a style property on the element and I hate that.  You could use a state instead, like const [overlayHidden, setOverlayHidden] = useState(false);  And the on the div use style={{ display: overlayHidden ? ‘none’ : ‘block’ }} or something like that.